src/shims/unix/foreign_items.rs

```diff
@@ -112,26 +112,18 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         match link_name.as_str() {
             // Environment related shims
             "getenv" => {
-                let [name] = this.check_shim_abi(
-                    link_name,
-                    abi,
-                    ExternAbi::C { unwind: false },
-                    [this.machine.layouts.const_raw_ptr.ty],
-                    this.machine.layouts.mut_raw_ptr.ty,
-                    args,
-                )?;
+                let (callee_abi, arg_types, ret_type) =
+                    shim_sig!(this, extern "C" fn((*const _)) -> (*const _));
+                let [name] =
+                    this.check_shim_abi(link_name, abi, callee_abi, arg_types, ret_type, args)?;
```

Comment:
Outsider question: How many places is `check_shim_abi` used so far? Could abi, args, and return get packaged into one argument so you could just use the macro directly in the parameters?